lib/spack/spack/fetch_strategy.py

```diff
@@ -55,74 +55,113 @@
 import spack.error
 import spack.util.crypto as crypto
 import spack.util.pattern as pattern
+
+from spack.util.compression import decompressor_for, extension
 from spack.util.executable import which
 from spack.util.string import comma_or
 from spack.version import Version, ver
-from spack.util.compression import decompressor_for, extension
 
-
-#: List of all fetch strategies, created by FetchStrategy metaclass.
+#: List of all known fetch strategies
 all_strategies = []
 
 
-def _needs_stage(fun):
-    """Many methods on fetch strategies require a stage to be set
-       using set_stage().  This decorator adds a check for self.stage."""
+def fetcher(cls):
+    """Registers a fetch strategy
+
+    Args:
+        cls (FetchStrategy): fetcher class to be registered
+    """
+    all_strategies.append(cls)
+    return cls
+
+
+def has_expanded_tree(directory):
```

Comment:
The `has_` prefix should be for things that actually return `bool`.  This doesn't -- I would call this `expanded_tree`.  It's much more legible to use that in a `bool` context than the other way around.